### PR TITLE
Lmswithreact

### PR DESCRIPTION
**Pull request by anuragsinghbisht**
I have added react and redux framework to the LMS project with updated webpack dev flow.

Note ** Please change the id in config.js to the contract id being used at the time.

### CODE REVIEW

File: app/actions/libraryActions.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+// Import libraries we need.
+import { default as Web3} from 'web3';
+import { default as contract } from 'truffle-contract'
+
+// Import our contract artifacts and turn them into usable abstractions.
+import lms_artifacts from '../../build/contracts/LMS.json'
+
+import contact from '../config';
```
Comment (nrchandan) [inline app/actions/libraryActions.js]: Should it be contract instead of contact? Does "../config" refer to app/config.js?

**Comment by nrchandan** [comment]
Why is the build failing? I just created a pull request and that has passed: https://github.com/nrchandan/lms/pull/7

File: app/actions/libraryActions.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            return instance.getAllBooks.call();
+        }).then((books) => {
+            dispatch(getAllBooksSuccess(books));
+            dispatch(getAllBooksLoading(false));
+        }).catch((e)=>{
+            dispatch(getAllBooksError(e));
+            dispatch(getAllBooksLoading(false));
+        });
+    }
+}
```
Comment (nrchandan) [inline app/actions/libraryActions.js]: I am sure you noticed a repetitive pattern here. Any way to avoid them?

File: app/web3.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Web3 } from 'web3';
+import { contract } from 'truffle-contract';
+
+if(typeof web3 !== 'undefined') {
+    window.web3 = new Web3(web3.currentProvider);
+} else {
+    console.warn("No web3 detected. Falling back to http://localhost:8545. You should remove this fallback when you deploy live, as it's inherently insecure. Consider switching to Metamask for development. More info here: http://truffleframework.com/tutorials/truffle-and-metamask");
+    window.web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
+}
```
Comment (nrchandan) [inline app/web3.js]: This violates the DRY principle. Saw a similar code in LibraryActions.js

File: test/app/components/Book.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+describe('Book',() => {
+
+  //Test : Components and its child renders without crashing
+  it('renders without crashing' , () => {
+    mount(<Book books="" />)
+  });
+
+  describe('render',() => {
+    //Test : Component gets rendered individually
+    it('should render the book',() => {
```
Comment (nrchandan) [inline test/app/components/Book.test.js]: You need to add a book first?

File: webpack.config.js [hunk 5]
```diff
@@ -1,36 +1,33 @@
...
+      { test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/ },
+      { test: /\.css$/, loader: ExtractTextPluginConfig.extract({
+        fallback : 'style-loader',
+        use : 'css-loader'
+       })}
     ]
-  }
-}
+  },
+  plugins : [ HTMLWebpackPluginConfig, ExtractTextPluginConfig]
```
Comment (nrchandan) [inline webpack.config.js]: Inconsistent spacing. Plan to start using linter (try npm run lint).

File: app/actions/libraryActions.js (hunk 1 above)
Comment (anuragsinghbisht) [inline app/actions/libraryActions.js]: Yes it should be contract and ref to app/config.js. I will fix it.

File: app/web3.js (hunk 3 above)
Comment (anuragsinghbisht) [inline app/web3.js]: This is a redundant code. I created it earlier but wasn't using it anywhere. I have removed it now.

File: webpack.config.js (hunk 5 above)
Comment (anuragsinghbisht) [inline webpack.config.js]: I have committed my code removing inconsistent spacing.

File: test/app/components/Book.test.js (hunk 4 above)
Comment (anuragsinghbisht) [inline test/app/components/Book.test.js]: This code was to check if the book component of react is rendering properly. I have to create another component/functionality to add a new book.

**Pull request merged**